opentelemetry-api/src/opentelemetry/trace/propagation/__init__.py

```diff
@@ -13,22 +13,24 @@
 # limitations under the License.
 from typing import Optional
 
-from opentelemetry import trace as trace_api
 from opentelemetry.context import get_value, set_value
 from opentelemetry.context.context import Context
+from opentelemetry.trace.span import INVALID_SPAN, Span
 
 SPAN_KEY = "current-span"
 
 
 def set_span_in_context(
```

Comment:
I'm wondering if this should be exposed in the trace api module as well? Would this ever be something a user set themselves?

Reply:
these are exposed via the trace_api. The reason they are not imported from there in this module is because it would create a cycle dependency.

A majority of my refactor here was to avoid the cyclic dependency introduced by re-exporting get_current_trace from the propagator.

Reply:
I'm a bit confused, what do you mean when you say "these are exposed via the trace_api"? How is this achieved if the function is not available in the api module?

Reply:
> I'm a bit confused, what do you mean when you say "these are exposed via the trace_api"? How is this achieved if the function is not available in the api module?

Sorry you're right. I can add it.

Reply:
added.